Sporadic R null pointer errors in tests

[R tests failed](https://github.com/opendp/opendp/actions/runs/8085390731/job/22093043633?pr=1177) with the following error, but no error when re-running. Tests should be deterministic.
```
══ Failed ══════════════════════════════════════════════════════════════════════
── 1. Error ('test-mod.R:8:34'): binary search ─────────────────────────────────
Error: Error in `make_chain_mt(rhs, lhs)`: [FFI] : null pointer: measurement1
Error: Error: Failures detected.
Execution halted
disabled backtrace
Backtrace:
     ▆
  1. ├─opendp::binary_search_chain(...) at test-mod.R:8:5
  2. │ ├─opendp (local) make_chain(...) at opendp/R/mod.R:507:3
  3. │ │ └─opendp::then_laplace(t_sum, s) at test-mod.R:8:34
  4. │ │   ├─opendp:::new_constructor_log(...) at opendp/R/measurements.R:1032:5
  5. │ │   │ └─opendp::new_hashtab(...) at opendp/R/mod.R:431:3
  6. │ │   └─opendp::new_hashtab(list("scale", "QO"), list(scale, .QO)) at opendp/R/mod.R:405:3
  7. │ └─opendp::binary_search_param(...) at opendp/R/mod.R:405:3
  8. │   └─opendp::binary_search(...) at opendp/R/mod.R:525:3
  9. │     └─opendp (local) predicate(mid) at opendp/R/mod.R:584:5
 10. │       └─make_chain(param)(d_in = d_in, d_out = d_out) at opendp/R/mod.R:526:5
 11. │         └─opendp::measurement_check(ptr, d_in, d_out) at opendp/R/mod.R:155:9
 12. │           └─opendp::measurement_input_distance_type(measurement) at opendp/R/core.R:54:5
 13. │             ├─opendp:::new_constructor_log(...) at opendp/R/core.R:133:5
 14. │             │ └─opendp::new_hashtab(...) at opendp/R/mod.R:431:3
 15. │             └─opendp::new_hashtab(list("this"), list(this)) at opendp/R/mod.R:405:3
 16. └─opendp::make_chain_mt(rhs, lhs) at opendp/R/mod.R:405:3

══ DONE ════════════════════════════════════════════════════════════════════════
Error: Process completed with exit code 1.
```

--- a/R/opendp/src/convert_elements.c
+++ b/R/opendp/src/convert_elements.c
@@ -112,7 +112,6 @@ SEXP anymeasurementptr_to_sexp(AnyMeasurement *input, SEXP log)
 {
     SEXP XPtr = PROTECT(R_MakeExternalPtr(input, AnyMeasurement_tag, R_NilValue));
     R_RegisterCFinalizerEx(XPtr, odp_AnyMeasurement_finalizer, TRUE);
-    UNPROTECT(1);
 
     int errorOccurred;
     SEXP new_measurement = PROTECT(get_private_func("new_measurement"));
@@ -121,7 +120,7 @@ SEXP anymeasurementptr_to_sexp(AnyMeasurement *input, SEXP log)
     if (errorOccurred)
         error("failed to construct measurement");
 
-    UNPROTECT(1);
+    UNPROTECT(2);
     return measurement;
 }
 

